File: selenium/src/main/java/pages/RegisterPage.java
package pages;

import base.BasePage;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class RegisterPage extends BasePage {

    // Locators
    private final By fullNameInput = By.cssSelector("input[type='text']");
    private final By emailInput = By.cssSelector("input[type='email']");
    private final By passwordInput = By.name("password");
    private final By confirmPasswordInput = By.name("confirmPassword");
    private final By termsCheckbox = By.cssSelector("input[type='checkbox']");
    private final By submitButton = By.cssSelector("button[type='submit']");

    public RegisterPage(WebDriver driver) {
        super(driver);
    }

    public void register(String fullName, String email, String password) {
        type(fullNameInput, fullName);
        type(emailInput, email);
        type(passwordInput, password);
        type(confirmPasswordInput, password);
        click(termsCheckbox);
        click(submitButton);
    }
}

File: e2e/pages/RegisterPage.js
export class RegisterPage {
  /**
   * @param {import('@playwright/test').Page} page
   */
  constructor(page) {
    this.page = page;
    this.fullNameInput = page.locator('input[type="text"]');
    this.emailInput = page.locator('input[type="email"]');
    this.passwordInput = page.locator('input[name="password"]');
    this.confirmPasswordInput = page.locator('input[name="confirmPassword"]');
    this.termsCheckbox = page.locator('input[type="checkbox"]');
    this.submitButton = page.locator('button[type="submit"]');
  }

  async goto() {
    await this.page.goto('/register');
  }

  async register(fullName, email, password) {
    await this.fullNameInput.fill(fullName);
    await this.emailInput.fill(email);
    await this.passwordInput.fill(password);
    await this.confirmPasswordInput.fill(password);
    await this.termsCheckbox.check();
    await this.submitButton.click();
  }
}
